cmd/soroban-cli/src/config/key.rs

```diff
@@ -0,0 +1,148 @@
+use std::{fmt::Display, str::FromStr};
+
+use serde::{Deserialize, Serialize};
+
+use super::secret::{self, Secret};
+use crate::xdr;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Failed to extract public key from secret")]
+    SecretPublicKey,
+    #[error(transparent)]
+    Secret(#[from] secret::Error),
+    #[error(transparent)]
+    StrKey(#[from] stellar_strkey::DecodeError),
+    #[error("failed to parse key {0}")]
+    Parse(String),
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum Key {
+    Secret(Secret),
+    PublicKey { public_key: Public },
+    MuxedAccount { muxed_account: MuxedAccount },
```

Comment:
Why the different patterns here where enum tuple variant is used sometime and struct variant is used in the other times?

Reply:
It is because of how I'm using serde for serialization, e.i. the field names (e.g. `public_key`) determine the case. `Secret` is itself an enum of structs. So if you prefer I could make an enum `Public` which serves the same.

Reply:
That's because we're using `#[serde(untagged)]`. If we used the default, they should be unambiguously tagged which would be the more common way to deal with this issue I think?

Reply:
Fixed it! Should change `Secret` too. But could be in separate PR.